packages/react-instantsearch/src/components/SearchBox.js

```diff
@@ -134,11 +136,19 @@ class SearchBox extends Component {
 
   onChange = e => {
     this.setQuery(e.target.value);
+
+    if (this.props.onChange) {
+      this.props.onChange(e);
```

Comment:
Yes, but React automatically dispose listeners when the DOM node is removed. If you do this by your own you will need to `removeEventListener` but its not recommended, let React do the things 👍